src/Razor/src/Microsoft.VisualStudio.RazorExtension/SyntaxVisualizer/SyntaxVisualizerControl.xaml.cs

```diff
@@ -155,6 +158,33 @@ public void ShowGeneratedHtml()
         OpenVirtualDocuments<HtmlVirtualDocumentSnapshot>(_activeWpfTextView.TextBuffer);
     }
 
+    public void ShowSerializedTagHelpers(TagHelperDisplayMode displayKind)
+    {
+        ThreadHelper.ThrowIfNotOnUIThread();
+
+        var codeDocument = GetCodeDocument();
+        var tagHelpers = displayKind switch
+        {
+            TagHelperDisplayMode.All => codeDocument.GetTagHelpers(),
+            TagHelperDisplayMode.InScope => codeDocument.GetTagHelperContext().TagHelpers,
+            TagHelperDisplayMode.Referenced => (IEnumerable<TagHelperDescriptor>)codeDocument.GetReferencedTagHelpers(),
```

Comment:
I imagine the API review for this went something like this:

> "We need a place to stor..."
> "Code document."
> "Okay, but that will clash with the naming of.."
> "Code. Document."


<sub>lol, like there was an API review</sub>